app/controllers/concerns/hyrax/works_controller_behavior.rb

```diff
@@ -377,6 +365,26 @@ def after_create_response
       end
     end
 
+    def after_create_error(errors, original_input_params_for_form = nil)
+      respond_to do |wants|
+        wants.html do
+          flash[:error] = errors.to_s
+          rebuild_form(original_input_params_for_form) if original_input_params_for_form.present?
```

Comment:
Rebuilding the form from the original input parameters only works for AF works.  The changeset retains the edited values when it is used to render the edit form after an error.